radix-engine/src/track/interface.rs

```diff
@@ -208,17 +208,41 @@ pub enum StoreAccess {
     ReadFromDb(CanonicalSubstateKey, usize),
     /// Non-existent substate was read from database.
     ReadFromDbNotFound(CanonicalSubstateKey),
-    /// A new entry has been added to track
-    /// System limits how many items that can be tracked.
-    NewEntryInTrack(CanonicalSubstateKey, usize),
+
+    /// A substate in track has been updated
+    UpdateSubstateInTrack {
+        /// The canonical substate key
+        canonical_substate_key: CanonicalSubstateKey,
+        /// Previous size of the substate, or `None` if it's a new entry
+        old_size: Option<usize>,
```

Comment:
Interesting! Can you add a comment that `old_size` refers to the current size before the update rather than the size in the underlying store. That was a little confusing to me.